src/config.cpp

```diff
@@ -1076,17 +1076,19 @@ class SimulationConfig::Parser
         return result;
     }
 
-    ConnectionMap parseConnectionOverrides() const {
-        ConnectionMap result;
+    std::vector<ConnectionOverride> parseConnectionOverrides() const {
+        std::vector<ConnectionOverride> result;
 
         const auto connIt = _json.find("connection_overrides");
-        if (connIt == _json.end())
+        if (connIt == _json.end()) {
             return result;
+        }
 
         for (auto it = connIt->begin(); it != connIt->end(); ++it) {
-            auto& connect = result[it.key()];
-            auto& valueIt = it.value();
-            const auto debugStr = fmt::format("connection_override {}", it.key());
+            const auto& valueIt = it.value();
+            ConnectionOverride connect;
+            parseMandatory(valueIt, "name", "connection_override", connect.name);
```

Comment:
Yes, great point, I will add that.

Reply:
Would it be worth to still detect we get a map to raise a more informative warning/error? Without going through a deprecation phase, maybe that's a good compromise.

Reply:
What sort of message are you suggesting?  I'd prefer to skip a deprecation stage, and people just change their configs.

Reply:
As a simple idea.
```
      if (connIt->is_object()) {
            throw SonataError("`connection_overrides` spec changed and must be specified as an array. Please update your config");
        }
```
Whether we temporarily support the old format or not, Im a bit neutral. It would allow us to move more smoothly in neurodamus, but won't be very problematic either way.

Reply:
I thought of supporting the old format, but I gave up in the end. As the python function signature is changed from `connection_override(name)` to `connection_overrides()`, neurodamus will change anyway to read the whole list of connections from libsonata. Considering the order is not guaranteed when parsing a dict, the simulation results are not guaranteed.

Reply:
> As a simple idea.

We're emitting the error `SonataError("`connection_overrides` must be an array")`, which I hope points people in the right direction.